common/scripts/kind_provisioner.sh

```diff
@@ -98,42 +98,54 @@ function cleanup_kind_cluster() {
   fi
 }
 
+function check_default_cluster_yaml() {
+  if [[ -z "${DEFAULT_CLUSTER_YAML}" ]]; then
+    echo 'DEFAULT_CLUSTER_YAML file must be specified. Exiting...'
+    return 1
+  fi
+}
+
 # setup_kind_cluster creates new KinD cluster with given name, image and configuration
 # Even in case of single cluster, don't call this directly. Call through setup_kind_clusters
 # with cluster topology configuration file.
 # 1. CLUSTER_CONFIG: KinD cluster configuration YAML file (mandatory)
 # 2. NAME: Name of the Kind cluster (optional)
 # 3. IMAGE: Node image used by KinD (optional)
-# 4. IP_FAMILY: valid values are ipv4 and ipv6.
 # This function returns 0 when everything goes well, or 1 otherwise
 # If Kind cluster was already created then it would be cleaned up in case of errors
 function setup_kind_cluster() {
-  CLUSTER_CONFIG_YAML=${1}
-  NAME="${2:-istio-testing}"
-  IMAGE="${3:-kindest/node:v1.18.2}"
-  IP_FAMILY="${4:-ipv4}"
+  NAME="${1:-istio-testing}"
+  IMAGE="${2:-kindest/node:v1.18.2}"
+  CONFIG="${3:-}"
+
+  check_default_cluster_yaml
 
   # Delete any previous KinD cluster
   echo "Deleting previous KinD cluster with name=${NAME}"
   if ! (kind delete cluster --name="${NAME}" -v9) > /dev/null; then
     echo "No existing kind cluster with name ${NAME}. Continue..."
   fi
 
-  # Patch cluster configuration if IPv6 is required
-  if [ "${IP_FAMILY}" = "ipv6" ]; then
-    grep 'ipFamily: ipv6' "${CLUSTER_CONFIG}" || \
-    cat <<EOF >> "${CLUSTER_CONFIG}"
+  # explicitly disable shellcheck since we actually want $NAME to expand now
+  # shellcheck disable=SC2064
+  trap "cleanup_kind_cluster ${NAME}" EXIT
+
+    # If config not explicitly set, then use defaults
+  if [[ -z "${CONFIG}" ]]; then
+      # Kubernetes 1.15+
+      CONFIG=${DEFAULT_CLUSTER_YAML}
+      # Configure the cluster IP Family only for default configs
+    if [ "${IP_FAMILY}" = "ipv6" ]; then
```

Comment:
Should this be done even if `CONFIG` is specified?

Reply:
Same as in master branch - https://github.com/istio/istio/blob/master/prow/lib.sh#L155
I had changed, but put that back after I had issues with `ipv6` cluster setup. Not sure, why it was like that. Maybe the rationale is "if cluster configuration is explicitly given then all these issues are already taken care of"?